### DISCUSSION TITLE
Fix Czech language diacritics

### DISCUSSION THREAD

**Discussion by pohlpiano** [PaddlePaddle/PaddleOCR#13914] [Ideas & Features]
Hi, this is an awesome project. I first tested with Chinese/English subtitles and was quite impressed. However, testing the Czech language, I found a weird thing - it comes out quite well, except for one significant detail: all diacritical marks come out with an umlaut like in German ü ö, etc., and no correct diacritics for Czech letters like ě,š,č,ř,ž.... Could anyone fix this, please?

**Comment by GreatV** [comment]
How did you test Czech lang? Can you provide a demo?

**Reply by pohlpiano** [reply]
I followed a guide from YouTube and I did exactly the same process, like with the Chinese/English bilingual subs. I tested with two versions of the video, one of them with yellow subtitles on solid black background, another with white subtitles on transparent background, the results pretty much the same. Most of special characters coming out with the "umlaut" mark, some of them with regular accent (long vowel) mark, but no small hooks, etc...

**Reply by pohlpiano** [reply]
Actually, it looks like most of the hooks excerpt for "ě" were just ignored, and it is mostly the little accents and little circles, which came out as umlauts...

3
00:00:17,936 --> 00:00:24,071
Dnes zacneme u lovcü mamutü
Bylo to v dobé kamenné a ledové

4
00:00:24,104 --> 00:00:27,005
Brrr, hroznä zima

...

00:01:27,483 --> 00:01:32,151
Doprava tuného ülovku vyžadovala
nejen silu, ale i chytrost.

11
00:01:32,184 --> 00:01:37,085
Zäkladem bylo tähnout mamuta
vzdy po srsti

12
00:01:58,356 --> 00:01:58,389
11

13
00:01:58,656 --> 00:02:01,956
Kolik dävate zazvoru,
milostiva?

**Comment by pohlpiano** [comment]
In detail, I found:
ě coming out as é
č coming out as regular c
š coming out as regular s /ONCE as correct š/
á coming out as ä
ů coming out as ü
ú coming out as ü
é CORRECTLY coming out as é
ě coming out as é
ž coming out as regular z
ř coming out as regular r
í coming out as regular i
ý coming out as regular y
č once ignored entirely
Č coming out as regular C

**Reply by pohlpiano** [reply]
The Czech video was a short one, three minutes, 24 subtitle lines.

**Reply by pohlpiano** [reply]
The subtitles are very clear 
![image](https://github.com/user-attachments/assets/635f36f2-9732-4d28-9f12-c051b54c19a3)

**Reply by pohlpiano** [reply]
Doba ledovä byla ideälni
pro zimni sporty.

**Reply by pohlpiano** [reply]
Should be: Doba ledová byla ideální pro zimní sporty.

**Comment by GreatV** [comment]
The Czech language is treated as a Latin language in processing, using the character dictionary from https://github.com/PaddlePaddle/PaddleOCR/blob/main/ppocr/utils/dict/latin_dict.txt. Check if it includes all Czech characters. If the dictionary is appropriate, you might need to fine-tune the OCR models for your specific scenario.

**Comment by pohlpiano** [comment]
Oh, It seems that some of these characters are missing (not all of them), indeed. How to fix that? Where exactly to add them?

**Reply by GreatV** [reply]
You need to complete the characters and retrain the model. For specific steps, please refer to https://paddlepaddle.github.io/PaddleOCR/en/ppocr/model_train/recognition.html#12-dictionary
